app/models/hackathon/digest/listings/by_location.rb

```diff
@@ -5,29 +5,48 @@ module Hackathon::Digest::Listings::ByLocation
   private
 
   def nearby_upcoming_hackathons
-    locations_to_search.flat_map do |location|
-      upcoming_hackathons.near(location, 50, units: :mi)
-    end.compact.uniq
-  end
+    subscriptions_to_search.flat_map do |subscription|
+      # Searching for hackathons **in** Subscription's location
+      hackathons = upcoming_hackathons.select do |hackathon|
+        hackathon.to_location.covered_by_or_equal?(subscription.to_location)
+      end
 
-  def locations_to_search
-    active_subscriptions = Hackathon::Subscription.active_for(recipient)
-    locations = active_subscriptions.collect(&:location).compact.uniq
+      # Searching for hackathons **near** Subscription's location
+      if subscription.to_location.sig_component == :city
+        nearby_hackathons = upcoming_hackathons
+          .near(subscription.location, 150, units: :mi)
+          .where.not(city: [nil, ""])
+
+        hackathons.concat(nearby_hackathons).uniq!
+      end
 
-    locations.sort_by! do |location| # sort by least specific to most specific location
-      location.split(", ").length
+      hackathons.map { |hackathon| {hackathon:, subscription:} }
```

Comment:
why are subscriptions being returned with this?

Reply:
`Hackathon::Digest::Listing` belongs_to `Hackathon::Subscription`. So we need to return both the hackathon and it's associated subscription to that a `Listing` can be created